syntax = "proto3"; // Define a versão do Protobuf usada
package usuarios;  // Nome do pacote do serviço

// Definição do serviço Usuarios, que contém os métodos disponíveis
service Usuarios {
  rpc ObterUsuarios (ObterUsuariosRequest) returns (ObterUsuariosResponse);   // Obtém todos os usuários
  rpc CriarUsuario (CriarUsuarioRequest) returns (CriarUsuarioResponse);  // Cria um novo usuário
}

// Estrutura de um usuário
message Usuario {
  string id = 1;       // Identificador único do usuário
  string nome = 2;     // Nome do usuário
  string email = 3;    // Endereço de e-mail do usuário
  string senha = 4;    // Senha do usuário
}

// Solicitação para obter todos os usuários (não precisa de parâmetros)
message ObterUsuariosRequest {}

message ObterUsuariosResponse {
  repeated Usuario usuarios = 1; // Lista de usuários retornada
}

// Solicitação para criar um novo usuário
message CriarUsuarioRequest {
  Usuario usuario = 1; // Dados do usuário a ser criado
}

message CriarUsuarioResponse {
  Usuario usuario = 1; // Usuário criado com sucesso
}


